timemachine/fe/free_energy.py

```diff
@@ -1472,10 +1475,14 @@ def sample_replica(xvb: CoordsVelBox, state_idx: StateIdx) -> tuple[NDArray, NDA
 
             current_step = current_frame * md_params.steps_per_frame
             # Setup the MC movers of the Context
+            starting_water_acceptances = 0
+            starting_water_proposals = 0
             for mover in context.get_movers():
                 if md_params.water_sampling_params is not None and isinstance(mover, WATER_SAMPLER_MOVERS):
                     assert water_params_by_state is not None
                     mover.set_params(water_params_by_state[state_idx])
+                    starting_water_proposals = mover.n_proposed()
+                    starting_water_acceptances = mover.n_accepted()
```

Comment:
Does this logic assume that there is only one water sampler in `context.get_movers()`? If so, should we assert this here?